pyfar/classes/coordinates.py

```diff
@@ -1374,16 +1374,347 @@ def show(self, mask=None, **kwargs):
+        points = coords.cartesian
+
+        # get KDTree
+        kdtree = self._make_kdtree()
+
+        # query nearest neighbors
+        points = points.flatten() if coords.csize == 1 else points
+
+        # nearest points
+        distance, index = kdtree.query(points, k=k)
+
+        if distance_measure == 'spherical':
+            # determine validate radius
+            radius = self.radius
+            delta_radius = np.max(radius) - np.min(radius)
+            if delta_radius > 1e-15:
+                raise ValueError(
+                    "find_nearest_sph only works if all points have the same \
+                    radius. Differences are larger than 1e-15")
+            radius = np.max(radius)
+
+            # convert cartesian coordinates to length on the great circle
+            distance = 2 * radius * np.arcsin(distance/(2*radius))
+
+        if self.cdim == 1:
+            if k > 1:
+                index_multi = np.moveaxis(index, -1, 0)
+                index = np.empty((k), dtype=tuple)
+                for kk in range(k):
+                    index[kk] = tuple([index_multi[kk]], )
+            else:
+                index = tuple([index], )
+        else:
+            index_array = np.arange(self.csize).reshape(self.cshape)
+            index_multi = []
+            for dim in range(self.cdim):
+                index_multi.append([])
+                for i in index.flatten():
+                    index_multi[dim].append(np.where(i == index_array)[dim][0])
+                index_multi[dim] = np.asarray(
+                    index_multi[dim]).reshape(index.shape)
+            if k > 1:
+                index_multi = np.moveaxis(index_multi, -1, 0)
+                index = np.empty((k), dtype=tuple)
+                for kk in range(k):
+                    index[kk] = tuple(index_multi[kk])
+            else:
+                index = tuple(index_multi)
+
+        if k > 1:
+            distance = np.moveaxis(distance, -1, 0)
+
+        return index, distance
+
+    def find_within(
+            self, coords, distance=0., distance_measure='euclidean',
+            atol=1e-15, return_sorted=False):
+        """
+        Find coordinates within certain distance to the query points.
+
+        Parameters
+        ----------
+        coords : pf.Coordinates
+            Coordinates to which the nearest neighbors are searched.
+        distance : number, optional
+            Maximum allowed distance to the given points ``coords``.
+            Distance must be >= 0. For just exact matches use ``0``.
+            The default is ``0``.
+        distance_measure : string, optional
+            ``'euclidean'``
+                distance is determined by the euclidean distance.
+                This is default.
+            ``'spherical'``
+                distance is determined by the great-circle distance.
+            ``'angular'``
+                distance is determined by the angles in radiant.
```

Comment:
Is this also great circle distance but in radians? Should we also add this option to `find_nearest`?

Reply:
See above, might be good to improve naming and documentation